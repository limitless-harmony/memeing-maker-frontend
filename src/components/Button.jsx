import React from 'react';
import styled from 'styled-components';

import { calculateRem } from 'styles';
import { white, black, darken } from 'styles/colors';

const Button = ({
  color = black,
  children,
  rounded,
  outline,
  fullWidth,
  onClick,
}) => {
  return (
    <StyledButton
      color={color}
      outline={outline}
      rounded={rounded}
      fullWidth={fullWidth}
      onClick={onClick}
    >
      {children}
    </StyledButton>
  );
};

export const StyledButton = styled.button`
  border: ${({ outline, color }) => (outline ? `2px solid ${color}` : 'none')};
  background-color: ${({ outline, color }) =>
    outline ? 'transparent' : color};
  color: ${({ outline, color }) => (outline ? color : white)};
  border-radius: ${({ rounded }) =>
    rounded ? calculateRem(4000) : calculateRem(8)};
  text-align: center;
  cursor: pointer;
  letter-spacing: 1px;
  width: ${({ fullWidth }) => (fullWidth ? '100%' : 'auto')};
  font-size: ${calculateRem(18)};
  margin: ${calculateRem(10)};
  padding: ${calculateRem(3)} ${calculateRem(30)};
  &:hover {
    background-color: ${({ color }) => darken(color, 0.05)};
    color: ${white};
  }
`;

export default Button;
