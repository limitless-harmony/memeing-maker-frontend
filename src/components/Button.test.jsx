import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import Button from 'components/Button';

describe('Testing the Button component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.find('button'));
    expect(wrapper).toMatchSnapshot();
  });
  it('It changes the background color', () => {
    const tree = renderer.create(<Button color="red" />).toJSON();
    expect(tree).toHaveStyleRule('background-color', 'red');
  });
});
