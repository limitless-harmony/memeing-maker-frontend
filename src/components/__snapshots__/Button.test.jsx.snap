// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Button component renders without crashing 1`] = `
<styled.button
  color="#000"
/>
`;
